uberfire-structure/uberfire-structure-backend/src/test/java/org/guvnor/structure/backend/pom/PomEditorDefaultTest.java

```diff
@@ -113,112 +186,131 @@ public void addDepTest() {
 
     @Test
     public void addDepsTest() {
-        DynamicPomDependency dep = new DynamicPomDependency("junit",
-                                                            "junit",
-                                                            "4.12",
-                                                            "");
-        List<DynamicPomDependency> deps = Arrays.asList(dep);
-        boolean result = editor.addDependencies(deps,
+        boolean result = editor.addDependencies(EnumSet.of(DependencyType.JPA),
                                                 PathFactory.newPath(tmp.toAbsolutePath().toString() + File.separator + POM,
                                                                     tmp.toUri().toString() + File.separator + POM));
         assertThat(result).isTrue();
     }
 
     @Test
     public void addDuplicatedDepTest() {
```

Comment:
@desmax74 Could you please add test where we try to add duplicate `junit` dependency but with different version.
E.g. there is a `junit:4.12` in pom but I want to add `junit:5.0`

Reply:
**User Scenarios**

- User is using `central`. He/she already have a project that runs smoothly, but user adds another persistable data-object. Suddenly his/hers project is not building anymore. If he/she checks pom.xml for JPA version, it is different than before (it is now changed to older version). Thing is that user decided to use newer `hibernate` version because he/she needs to, but it is overwritten when he/she creates new objects even if user does not want to.
> I suggest prompting user if he wants us to update the version in this case. So instead of overwriting the version we should let user make the decision and possibly remember his/hers decision

- User is upgrading from X.0 to X.1 version. His/hers projects (couple of hundreds) use `JPA` dependencies that have version older than our version in business-central. 
Do we update them or not? If we do, how do we update them? 
We might also break users projects without user explicitly knowing it until he tries to build them.

 @desmax74 @ederign thoughts? How should we handle these cases?

Reply:
Hi @domhanak,

I think the mismatch here is between 'regular java projects' and 'kie java projects'.

Our projects need to have a straight alignment with our versions because in the end, they will run in kie-server.

So, "user decided to use newer hibernate version because he/she needs to"
"I want to add junit:5.0"

He can't. His version needs to be aligned with ours.

"User is upgrading from X.0 to X.1 version"

For this, we can discuss in the future a migration tool.


My call would be to at least at first, manage all pom dependencies and update them if necessary.

@porcelli wdyt?